registry/remote_factory_test.go

```diff
@@ -123,7 +123,59 @@ func TestRemoteFactory_CredentialsFromConfigHTTPSHosts(t *testing.T) {
 	if pass != c.password {
 		t.Fatalf("Expected %q, got %q.", pass, c.password)
 	}
-	if len(creds.Hosts()) != 1 || httpsHost != creds.Hosts()[0] {
+	if len(creds.Hosts()) != 1 || host != creds.Hosts()[0] {
 		t.Fatalf("Expected %q, got %q.", httpsHost, creds.Hosts()[0])
 	}
 }
+
+func TestRemoteFactory_ParseHost(t *testing.T) {
+	user := "user"
+	pass := "pass"
+
+	for _, v := range []struct {
+		host        string
+		imagePrefix string
+	}{
+		{
+			host:        "host",
+			imagePrefix: "host",
+		},
+		{
+			host:        "gcr.io",
+			imagePrefix: "gcr.io",
+		},
+		{
+			host:        "https://gcr.io",
+			imagePrefix: "gcr.io",
+		},
+		{
+			host:        "https://gcr.io/v1",
+			imagePrefix: "gcr.io",
+		},
+		{
+			host:        "https://gcr.io/v1/",
```

Comment:
Should probably also test some malformed ones, and what happens...
`"https://"`
`""`
`"/usr/local"`